src/history/patient-history.test.tsx

```diff
@@ -0,0 +1,151 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import { useParams } from 'react-router-dom';
+import HistoryDetailOverview from './history-detail-overview.component';
+import React from 'react';
+import { FetchResponse, usePatient } from '@openmrs/esm-framework';
+import { createIpsResource, useIpsResource } from '../common/history.resource';
+import { mockIPS, mockPUTResponse } from '../test.util';
+
+jest.mock('../common/history.resource');
+
+jest.mock('react-router-dom', () => ({
+  ...(jest.requireActual('react-router-dom') as any),
+  useLocation: jest.fn().mockReturnValue({
+    pathname: 'openmrs/spa/patient',
+  }),
+  useHistory: () => [],
+  useParams: jest.fn().mockReturnValue({ patientUuid: 'undefined' }),
+}));
+
+describe('Display patient IPS', () => {
+  beforeEach(() => {
+    const mockUseParams = useParams as jest.Mock;
+    mockUseParams.mockReturnValue({ patientUuid: 'mock-uuid' });
+  });
+
+  it('renders without crashing', () => {
+    const mockUsePatient = jest.mocked(usePatient);
+    const mockUseIpsResource = jest.mocked(useIpsResource);
```

Comment:
Why not create the mocks once, outside of the `it()` block and then just change the return value?